Build failure: isl_0_20

### Steps To Reproduce

Steps to reproduce the behavior:
1. build isl-0.20 on riscv64-linux

### Build log

```
building '/nix/store/q7c67kcd89yymn222igjz58dlgaij489-isl-0.20.drv'...
unpacking sources
unpacking source archive /nix/store/x9v0hmr76dm68pijzldzxks01iqzag1z-isl-0.20.tar.xz
source root is isl-0.20
setting SOURCE_DATE_EPOCH to timestamp 1532811285 of file isl-0.20/interface/isl.py
patching sources
configuring
fixing libtool script ./ltmain.sh
./configure
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/9hkiw8kgbi3zcclxk3l8jh624gin57pj-isl-0.20 --with-gcc-arch=generic
checking for a BSD-compatible install... /nix/store/am27aaylz23yrws5bl5v858bg57qjdg3-bootstrap-tools/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /nix/store/am27aaylz23yrws5bl5v858bg57qjdg3-bootstrap-tools/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
./configure: line 3487: cd: /nix/store/9hkiw8kgbi3zcclxk3l8jh624gin57pj-isl-0.20: No such file or directory
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking for style of include used by make... GNU
checking dependency style of gcc... none
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... none
checking for C compiler vendor... gnu
checking build system type... ./config.guess: unable to guess system type

This script, last modified 2014-03-23, has failed to recognize
the operating system you are using. It is advised that you
download the most up to date version of the config scripts from

  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.guess;hb=HEAD
and
  http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=HEAD

If the version you run (./config.guess) is already up to date, please
send the following data and any information you think might be
pertinent to <config-patches@gnu.org> in order to provide the needed
information to handle your system.

config.guess timestamp = 2014-03-23

uname -m = riscv64
uname -r = 5.15.0
uname -s = Linux
uname -v = #1-NixOS SMP Mon Mar 20 17:53:42 UTC 2023

/usr/bin/uname -p =
/bin/uname -X     =

hostinfo               =
/bin/universe          =
/usr/bin/arch -k       =
/bin/arch              =
/usr/bin/oslevel       =
/usr/convex/getsysinfo =

UNAME_MACHINE = riscv64
UNAME_RELEASE = 5.15.0
UNAME_SYSTEM  = Linux
UNAME_VERSION = #1-NixOS SMP Mon Mar 20 17:53:42 UTC 2023
configure: error: cannot guess build type; you must specify one
error: builder for '/nix/store/q7c67kcd89yymn222igjz58dlgaij489-isl-0.20.drv' failed with exit code 1

```

### Additional context

isl 0.20 is ancient, and needs a config.guess update I guess

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
(fails to run due to above error)
```

--- a/pkgs/development/libraries/isl/generic.nix
+++ b/pkgs/development/libraries/isl/generic.nix
@@ -5,7 +5,7 @@
 , patches ? []
 }:
 
-{ lib, stdenv, fetchurl, gmp
+{ lib, stdenv, fetchurl, gmp, autoreconfHook
 }:
 
 stdenv.mkDerivation {
@@ -19,6 +19,7 @@ stdenv.mkDerivation {
   inherit patches;
 
   strictDeps = true;
+  nativeBuildInputs = lib.optionals (stdenv.hostPlatform.isRiscV && lib.versionOlder version "0.24") [ autoreconfHook ];
   buildInputs = [ gmp ];
 
   inherit configureFlags;

